Sentinel/Classes/UserDefaults/UserDefaultsToolDetailView.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  UserDefaultsToolDetailView.swift
+//  Sentinel
+//
+//  Created by Zvonimir Medak on 28.11.2024..
+//
+
+import SwiftUI
+
+struct UserDefaultsToolDetailView: View {
+
+    @Environment(\.presentationMode) var presentationMode: Binding<PresentationMode>
+    @ObservedObject var viewModel: UserDefaultsToolDetailViewModel
+    #if os(macOS)
+    @Binding var selection: String?
+    #endif
+
+    var body: some View {
+        VStack(spacing: 10) {
+            Text(viewModel.value)
+                .contextMenu(ContextMenu(menuItems: {
+                    Button("Copy", action: {
+                        #if os(macOS)
+                        NSPasteboard.general.setString(viewModel.value, forType: .string)
+                        #else
+                        UIPasteboard.general.string = viewModel.value
+                        #endif
+                    })
+                }))
+                .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .topLeading)
+
+            Button(action: {
+                viewModel.didPressDelete()
+                #if os(macOS)
+                selection = nil
+                #else
+                presentationMode.wrappedValue.dismiss() // on macOS dismisses the whole window which isn't desired
```

Comment:
I've seen this used multiple times; can we instead make an extension that will handle this case by calling it as a one liner on a calling side and define inner logic in the extension?

Reply:
To extract this into a protocol or something we would still have to have if/else since the selection is hidden behind an if macos, and we would have to provide it somehow. It's called twice in the codebase, so for now, maybe we can leave it as is since macOS might be upped to version 13 so we can use NavigationStack